runtime/compiler/env/VMJ9.cpp

```diff
@@ -4086,6 +4090,30 @@ TR_J9VMBase::canDereferenceAtCompileTime(TR::SymbolReference *fieldRef, TR::Comp
       return false;
    }
 
+bool
+TR_J9VMBase::isStable(TR::Symbol *field, int cpIndex, TR_ResolvedMethod *owningMethod, TR::Compilation *comp)
```

Comment:
This method needs to be overridden for AOT and JIT server.
- AOT (`TR_J9SharedCacheVM`): true results need to be validated on load, or we need to validate something else that guarantees the result would be true, e.g. the class chain of the declaring class
- JIT server (`TR_J9ServerVM`): `fieldClass` is a remote pointer and we're not in the right process to call `jitIsFieldStable()`

There's also the combination of both at once (`TR_J9SharedCacheServerVM`)

To get this going without needing to make it work in AOT and JIT server in one fell swoop, it should suffice to just return false in those other subclasses, since treating fields as non-stable is conservative. (`TR_J9SharedCacheServerVM` could inherit that implementation from `TR_J9ServerVM`, but the implementations would need to diverge when support for stable fields is implemented later)

Reply:
Yeah, very good point! I will fix it.